packages/alert-dialog/src/AlertDialog.ts

```diff
@@ -0,0 +1,230 @@
+        );
+        if (ids.length) {
+            this.conditionLabelledby = conditionAttributeWithId(
+                this,
+                'aria-labelledby',
+                ids
+            );
+        }
+    }
+
+    public shouldManageTabOrderForScrolling = (): void => {
+        const { offsetHeight, scrollHeight } = this.contentElement;
+        if (offsetHeight < scrollHeight) {
+            this.contentElement.tabIndex = 0;
+        } else {
+            this.contentElement.removeAttribute('tabindex');
+        }
+    };
+
+    private describedbyId = `sp-dialog-description-${AlertDialog.instanceCount++}`;
+
+    protected onContentSlotChange({
+        target,
+    }: Event & { target: HTMLSlotElement }): void {
+        requestAnimationFrame(() => {
+            // Content must be available _AND_ styles must be applied.
+            this.shouldManageTabOrderForScrolling();
+        });
+        if (this.conditionDescribedby) {
+            this.conditionDescribedby();
+            delete this.conditionDescribedby;
+        }
+        const ids = gatherAppliedIdsFromSlottedChildren(
+            target,
+            this.describedbyId
+        );
+        if (ids.length && ids.length < 4) {
+            this.conditionDescribedby = conditionAttributeWithId(
+                this,
+                'aria-describedby',
+                ids
+            );
+        } else if (!ids.length) {
+            const idProvided = !!this.id;
+            if (!idProvided) this.id = this.describedbyId;
+            const conditionDescribedby = conditionAttributeWithId(
+                this,
+                'aria-describedby',
+                this.id
+            );
+            this.conditionDescribedby = () => {
+                conditionDescribedby();
+                if (!idProvided) {
+                    this.removeAttribute('id');
+                }
+            };
+        }
+    }
+
+    protected renderButtons(): TemplateResult {
+        return html`
+            <sp-button-group class="button-group">
+                <slot name="button"></slot>
+            </sp-button-group>
+        `;
+    }
+
+    protected override render(): TemplateResult {
+        return html`
+            <div class="grid">
+                <div class="header">
+                    ${this.renderHeading()} ${this.renderIcon()}
+                </div>
+                <sp-divider size="m" class="divider"></sp-divider>
+                ${this.renderContent()} ${this.renderButtons()}
+            </div>
+        `;
+    }
+
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        window.addEventListener(
+            'resize',
+            this.shouldManageTabOrderForScrolling
+        );
+    }
+
+    public override disconnectedCallback(): void {
+        window.removeEventListener(
+            'resize',
+            this.shouldManageTabOrderForScrolling
+        );
+        super.disconnectedCallback();
+    }
```

Comment:
Could I get you to volunteer for a follow up change that moved this to a Resize Observer so that we can also fix https://github.com/adobe/spectrum-web-components/issues/3613?

Reply:
Sure!